rr_iarrc/src/laplacian_line_detection/laplacian_line_detector.cpp

```diff
@@ -12,53 +12,58 @@
 #include <stdio.h>
 
 using namespace cv;
+using namespace std;
 
 cv_bridge::CvImagePtr cv_ptr;
 ros::Publisher pub, pub1, pub2, pub3;
 
 cv::Mat kernel(int, int);
 cv::Mat fillColorLines(cv::Mat, cv::Mat);
-void cutEnvironment(cv::Mat);
+void cutEnvironment(cv::Mat, int offset);
 cv::Mat cutSmall(cv::Mat, int);
 void publishMessage(ros::Publisher, Mat, std::string);
 Mat overlayBinaryGreen(Mat, Mat);
+Mat removeAngels(Mat img, int distanceFromEarth);
 
 int blockSky_height, blockWheels_height, blockBumper_height;
-int perfect_lines_min_cut, Laplacian_threshold;
+int perfect_lines_min_cut, Laplacian_threshold, adaptive_mean_threshold;
 
 
 void img_callback(const sensor_msgs::ImageConstPtr& msg) {
     cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
     Mat frame = cv_ptr->image;
 
-    //Image Height Debugger
-    //cutEnvironment(frame);
+    int orginalHeight = frame.rows;
+    int orginalWidth = frame.cols;
+    resize(frame, frame, Size(400, 400));
 
     cv::Mat frame_gray, frame_blur, detected_edges;
-    GaussianBlur(frame, frame_blur, Size(5,5), 0, 0, BORDER_DEFAULT);
+    GaussianBlur(frame, frame_blur, Size(5,5), 0);
     cv::cvtColor(frame_blur, frame_gray, cv::COLOR_BGR2GRAY);
 
     Mat thres;
     adaptiveThreshold(frame_gray, thres, 255, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, 5, -1);
-    cutEnvironment(thres);
-    Mat cut = cutSmall(thres, perfect_lines_min_cut);   //Make sure only Lines
+    cutEnvironment(thres, 0);
+    erode(thres, thres, kernel(3,1));
```

Comment:
Harping on importing the cv namespace again: my gut reaction here was to consult the OpenCV docs on cutEnvironment, but those don't exist because it's ours